core/src/main/java/io/grpc/LoadBalancerRegistry.java

```diff
@@ -89,27 +100,33 @@ public static synchronized LoadBalancerRegistry getDefaultRegistry() {
           HARDCODED_CLASSES,
           LoadBalancerProvider.class.getClassLoader(),
           new LoadBalancerPriorityAccessor());
-      instance = new LoadBalancerRegistry(providerList);
+      instance = new LoadBalancerRegistry();
+      for (LoadBalancerProvider provider : providerList) {
+        logger.fine("Service loader found " + provider);
+        if (provider.isAvailable()) {
+          instance.register(provider);
```

Comment:
addAll + refresh would be more ideal, to avoid the N^2 (not that that really matters given the sizes involved, except this is done at startup and will probably be interpreted).

Reply:
Done in a slightly different way.